pennylane/ops/op_math/controlled_ops.py

```diff
@@ -119,22 +120,61 @@ def _unflatten(cls, data, metadata):
             data[0], control_wires=metadata[0], control_values=metadata[1], work_wires=metadata[2]
         )
 
+    # pylint: disable=arguments-differ, too-many-arguments, unused-argument
+    @classmethod
+    def _primitive_bind_call(
+        cls,
+        base,
+        control_wires: WiresLike,
+        wires: WiresLike = (),
+        control_values=None,
+        unitary_check=False,
+        work_wires: WiresLike = (),
+    ):
+        wires = Wires(()) if wires is None else Wires(wires)
+        work_wires = Wires(()) if work_wires is None else Wires(work_wires)
+
+        if getattr(base, "wires", False) and len(wires) != 0:
+            warnings.warn(
+                "base operator already has wires; values specified through wires kwarg will be ignored."
+            )
+            wires = Wires(())
+
+        all_wires = control_wires + wires
+        return cls._primitive.bind(
+            base, control_wires=all_wires, control_values=control_values, work_wires=work_wires
+        )
+
     # pylint: disable=too-many-arguments,too-many-positional-arguments
     def __init__(
         self,
         base,
-        control_wires,
-        wires=None,
+        control_wires: WiresLike,
+        wires: WiresLike = (),
         control_values=None,
         unitary_check=False,
-        work_wires=None,
+        work_wires: WiresLike = (),
     ):
-        if getattr(base, "wires", False) and wires is not None:
+        wires = () if wires is None else Wires(wires)
+        work_wires = () if work_wires is None else Wires(work_wires)
+        control_wires = Wires(control_wires)
+
+        if getattr(base, "wires", False) and len(wires) != 0:
             warnings.warn(
                 "base operator already has wires; values specified through wires kwarg will be ignored."
             )
+            wires = Wires(())
 
         if isinstance(base, Iterable):
+            if len(wires) == 0:
+                if len(Wires(control_wires)) > 1:
+                    num_base_wires = int(qml.math.log2(qml.math.shape(base)[-1]))
+                    wires = Wires(control_wires)[-num_base_wires:]
+                    control_wires = Wires(control_wires)[:-num_base_wires]
+                else:
+                    raise TypeError(
+                        "Must specify a set of wires. None is not a valid `wires` label."
+                    )
```

Comment:
`TypeError`s are for when function arguments are for the wrong type, so a different exception is probably more appropriate ([ref](https://docs.python.org/3/library/exceptions.html#TypeError))

```suggestion
                    raise ValueError(
                        "Must specify a set of wires. None is not a valid `wires` label."
                    )
```

Reply:
Hmm ... the way I see it is that I am providing a set of wires of type `NoneType`, rather than `WiresLike`, hence the `TypeError`. 

Although, I can see how providing `wires=None` could be seen as the wrong value for a `Wires` object. Although, I'm not really set on which one is more appropriate lol.

Reply:
Fair enough, I'm happy to leave it as is.

Reply:
Resolving!